src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/PackageActionBaseCommand.cs

```diff
@@ -293,8 +293,23 @@ protected override void EndProcessing()
 
         protected async Task CheckPackageManagementFormat()
         {
-            // check if Project has any packages installed
-            if (!(await Project.GetInstalledPackagesAsync(Token)).Any())
+            bool packagesConfigAndSupportsPackageReferences = false;
+
+            if (Project.ProjectStyle == ProjectModel.ProjectStyle.PackagesConfig)
+            {
+                try
+                {
+                    packagesConfigAndSupportsPackageReferences = Project.ProjectServices.Capabilities.SupportsPackageReferences;
+                }
+                catch (NotImplementedException)
+                {
+                    //If Capabilities aren't implemented, then we assume it doesn't support Package References.
+                    return;
+                }
+            }
```

Comment:
Found another similar implementation where there is no try catch block. Can we follow this approach ?
https://github.com/NuGet/NuGet.Client/blob/64f2febf7d1de8b0228eca5afed97a96c5a30bba/src/NuGet.Clients/NuGet.PackageManagement.UI/Actions/UIActionEngine.cs#L624-L628

Reply:
It is similar and this is what I used to model my change. However, the UI and PowerShell plumbing feels a lot different to me. Let me explain my thought process, and I'd appreciate your input!

**PMUI** utilizes `UIActionEngine`
-> a Singleton Default provider is used, with a default Capabilities object, which it maps to a value of "false" for [SupportsPackageReferences](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.PackageManagement/Projects/DefaultProjectServices.cs#L37). 

To me, this provides some safety against an unhandled exception. 

**PMC** utilizes `PackageActionBaseCommand` which relies on VsSolutionManager to lookup the Project and provide a Default Project. This codepath is much different.

The path begins with [Preprocess() in GetPackageCommand](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Cmdlets/GetPackageCommand.cs#L97) which calls [`GetNuGetProjectAsync` in NuGetPowerShellBaseCommand](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/NuGetPowerShellBaseCommand.cs#L475) which calls `VsSolutionManager.GetDefaultNuGetProjectAsync`.

[VsSolutionManager](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs#L79) seems a bit less restrictive than the UIActionEngine. 
 -> we end up at [TryGetNuGetProject](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/ProjectSystemCache.cs#L45) which reads from the cache.

How did the cache get populated?
We call `AddProject` [here AddVsProjectAdapterToCacheAsync](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs#L713).
This `nuGetProject` is created by calling [ProjectSystemCache.AddProject](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/ProjectSystemCache.cs#L229). 

NuGetProjectFactory creates the project via [TryCreateNuGetProjectAsync](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/NuGetProjectFactory.cs#L63).

The [NetCorePackageReferenceProjectProvider creates projectServices](TryCreateNuGetProjectAsync) and these services can throw NotSupportExceptions ([Capabilities ](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectServices/NetCoreProjectSystemServices.cs#L31) is one of them).
**I actually need to add a NotSupportException catch block, now that I see this.**

//cc: @nkolev92

Reply:
Removed try-catch and now relying on the fact that "it's not possible to be PC and not implement Capabilities" with existing implementations.